### PR TITLE
Add terraform gcp module

### PR DESCRIPTION
**Pull request by alkar**

### CODE REVIEW

File: terraform/gcp/io.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+variable "role_name" {
+  type        = string
```
Comment (hectorhuertas) [inline terraform/gcp/io.tf:2]: why not use the default?

File: terraform/gcp/main.tf [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  zone = data.google_compute_zones.available.names[count.index]
+}
+
+resource "google_dns_record_set" "wiresteward" {
+  count = local.instance_count
+  name  = "${local.wireguard_endpoint[count.index]}."
+  type  = "A"
+  ttl   = 30 # TODO increase the value once happy with setup
```
Comment (hectorhuertas) [inline terraform/gcp/main.tf:52]: is this TODO left intentionally?

File: terraform/gcp/io.tf (hunk 1 above)
Comment (alkar) [reply terraform/gcp/io.tf:2]: How do you mean?

File: terraform/gcp/main.tf (hunk 2 above)
Comment (alkar) [reply terraform/gcp/main.tf:52]: Yes, once we're happy with the module we should set it to something more reasonable but the module is more than likely to change shape in the next couple weeks.

File: terraform/gcp/io.tf (hunk 1 above)
Comment (hectorhuertas) [reply terraform/gcp/io.tf:2]: Don't need to define type if setting a default, i think? Maybe even when not setting default

Comment (alkar) [reply terraform/gcp/io.tf:2]: Ah I see, you're right. Just being more explicit)

**Review by hectorhuertas — APPROVED** [review]

**Pull request merged**